lib/rules/no-global-jquery.js

```diff
@@ -28,96 +24,18 @@ module.exports = {
   ERROR_MESSAGE,
 
   create(context) {
-    let destructuredAssignment;
-    let importAliasName;
-    let hasJqueryImport;
-    let hasEmberImport;
-
     return {
-      ImportDeclaration(node) {
-        const validImportAlias = new Set(['ember', 'jquery']);
-        // Track if the 'jquery' and/or 'ember' module (if an application
-        // is not using the new modules import syntax) is being imported.
-        // Will exclude non-ember or non-jquery imports such as:
-        //    - `import Foo from 'bar';`
-        //    - `import { readOnly } from '@ember/object/computed';`
-        if (validImportAlias.has(node.source.value)) {
-          const isJqueryImport = node.source.value === 'jquery';
-          if (isJqueryImport) {
-            hasJqueryImport = true;
-            importAliasName = node.source.value;
-          } else {
-            const emberImport = ember.getEmberImportAliasName(node);
-            if (emberImport) {
-              hasEmberImport = true;
-              importAliasName = emberImport;
-            }
-          }
-        }
-      },
-
-      VariableDeclarator(node) {
-        // If an application is using an older version of Ember, not using
-        // the new modules import syntax, then lets locate a destructured
-        // jQuery assignment.
-        if (hasEmberImport) {
-          // Let's verify that we are dealing with an Ember MemberExpression
-          // only to test against.
-          if (
-            node.init &&
-            types.isMemberExpression(node.init) &&
-            isEmberMemberExpression(node.init)
-          ) {
-            const isJQueryVariable = node.init.property.name === '$';
+      'Program:exit'() {
```

Comment:
I've based this change off the work in https://github.com/eslint/eslint/blob/f6e7e3231bc43c989f8c953de8e0d328bac5eea0/lib/rules/no-undef.js#L58-L76

It checks the global scope to see if there is any reference using an identifier of `$` or `jQuery`

I did this while looking for things to clean up after changing the `no-jquery` rule. The only usage of `utils.isGlobalCallExpression` was in `no-global-jquery`, so I spent some time trying to remove it.

Again, I'm not really sure if this has any implications outside what's currently being tested, like performance issues or any missing test cases